src/frequenz/sdk/timeseries/logical_meter/_formula_engine.py

```diff
@@ -169,6 +169,16 @@ async def _run(self) -> None:
             else:
                 await sender.send(msg)
 
+    async def _stop(self) -> None:
+        """Stop a running formula engine."""
+        if self._task is None:
+            return
+        self._task.cancel()
+        try:
+            await self._task
+        except asyncio.CancelledError:
+            pass
```

Comment:
You can use that:
https://github.com/frequenz-floss/frequenz-sdk-python/blob/v0.x.x/src/frequenz/sdk/util/asyncio.py#L9

Reply:
done